CMakeLists.txt

```diff
@@ -118,81 +59,71 @@ else()
 	message( FATAL_ERROR "Boost library is not installed in your system" )    
 endif()
 
-
-# Create header file with the XX_ENABLE flags
-#set (HEADER_FILE "include/enable_flags.hpp")
-#file (WRITE ${HEADER_FILE} "//Enable Flags")
-
 # TBB Library 
-option( TBB_ENABLE "Require TBB library" ON )
-message( STATUS "TBB: " ${TBB_ENABLE} )
-if( TBB_ENABLE )
+option( GRPPI_TBB_ENABLE "Require TBB library" ON )
+message( STATUS "TBB: " ${GRPPI_TBB_ENABLE} )
+if( GRPPI_TBB_ENABLE )
     include_directories(${TBB_INCLUDE_DIRS} )
     find_package(TBB REQUIRED)
     add_definitions( "-DGRPPI_TBB" )
-#    file ( APPEND ${HEADER_FILE} "\n#define TBB_ENABLE ON")
-endif( TBB_ENABLE )
+endif( GRPPI_TBB_ENABLE )
 
 # Threads Library
 find_package(Threads REQUIRED)
 
 # OMP library
-option( OMP_ENABLE "Require OpenMP library" ON )
-message( STATUS "OpenMP: " ${OMP_ENABLE})
-if( OMP_ENABLE )
+option( GRPPI_OMP_ENABLE "Require OpenMP library" ON )
+message( STATUS "OpenMP: " ${GRPPI_OMP_ENABLE})
+if( GRPPI_OMP_ENABLE )
     find_package(OpenMP REQUIRED)
     add_definitions( "-DGRPPI_OMP" )
-#    file ( APPEND ${HEADER_FILE} "\n#define OMP_ENABLE ON")
-endif( OMP_ENABLE )
+    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}" )
+endif( GRPPI_OMP_ENABLE )
 
 # CUDA library
-option( CUDA_ENABLE "Require CUDA library" OFF )
+option( GRPPI_CUDA_ENABLE "Require CUDA library" OFF )
 message( STATUS "CUDA: " ${CUDA_ENABLE})
-if( CUDA_ENABLE )
+if( GRPPI_CUDA_ENABLE )
     
     if ( APPLE )
         message(WARNING "Apple NVCC no longer supported with gcc")
```

Comment:
I'm looking for documentation about this and looks like there are problems with clang and cuda for some versions.
Also the users comment that an additional library called xcode is required. 

I'm not sure how to approach this. Should we try to search for the latest tuple of clang and cuda version? should we consider the xcode library?